README.md

```diff
@@ -326,6 +326,59 @@ trainer.fit(pl_model, train_dataloaders=train_loader, val_dataloaders=val_loader
 </p>
 </details>
 
+If you want to train your model in the DDP regime (Distributed Data Parallel), you
+only need to slightly change 3 lines of code in the example below.
+
+<details>
+<summary>Training + Validation [Lightning Distributed]</summary>
+<p>
+
+[comment]:lightning-ddp-start
+```python
+import pytorch_lightning as pl
+import torch
+
+from oml.datasets.base import DatasetQueryGallery, DatasetWithLabels
+from oml.lightning.modules.retrieval import RetrievalModuleDDP
+from oml.lightning.callbacks.metric import MetricValCallbackDDP
+from oml.losses.triplet import TripletLossWithMiner
+from oml.metrics.embeddings import EmbeddingMetricsDDP
+from oml.miners.inbatch_all_tri import AllTripletsMiner
+from oml.models.vit.vit import ViTExtractor
+from oml.samplers.balance import BalanceSampler
+from oml.utils.download_mock_dataset import download_mock_dataset
+
+dataset_root = "mock_dataset/"
+df_train, df_val = download_mock_dataset(dataset_root)
+
+# model
+model = ViTExtractor("vits16_dino", arch="vits16", normalise_features=False)
+
+# train
+optimizer = torch.optim.SGD(model.parameters(), lr=1e-6)
+train_dataset = DatasetWithLabels(df_train, dataset_root=dataset_root)
+criterion = TripletLossWithMiner(margin=0.1, miner=AllTripletsMiner())
+batch_sampler = BalanceSampler(train_dataset.get_labels(), n_labels=2, n_instances=3)
+train_loader = torch.utils.data.DataLoader(train_dataset, batch_sampler=batch_sampler)
+
+# val
+val_dataset = DatasetQueryGallery(df_val, dataset_root=dataset_root)
+val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=4)
+metric_callback = MetricValCallbackDDP(metric=EmbeddingMetricsDDP())  # ddp specific
+
+# run
+pl_model = RetrievalModuleDDP(model=model, criterion=criterion, optimizer=optimizer,
+                              loaders_train=train_loader, loaders_val=val_loader  # ddp specific
+                              )
+trainer = pl.Trainer(max_epochs=1, callbacks=[metric_callback], num_sanity_val_steps=0,
+                     accelerator="auto", devices=2, strategy=pl.plugins.DDPPlugin(), replace_sampler_ddp=False  # ddp specific
```

Comment:
maybe group new parameters to some dict? To highlight exactly new parameters

Reply:
they are grouped in a new line already

Reply:
but okay